### ISSUE TITLE
Build error

### ISSUE THREAD

**Issue by max-sixty** [wangfenjin/duckdb-rs#187]
I'm trying to look at #178, but I get a basic compile error:

```
cargo build
   Compiling rustversion v1.0.0
   Compiling serde_json v1.0.0
   Compiling arrow-array v43.0.0
   Compiling regex v1.7.3
   Compiling regex-syntax v0.7.3
   Compiling arrayvec v0.5.0
   Compiling serde v1.0.85
   Compiling hashbrown v0.12.0
   Compiling fallible-streaming-iterator v0.1.0
   Compiling fallible-iterator v0.3.0
   Compiling smallvec v1.6.1
error[E0432]: unresolved import `num::complex::ComplexFloat`
  --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-array-43.0.0/src/arithmetic.rs:21:5
   |
21 | use num::complex::ComplexFloat;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ no `ComplexFloat` in `complex`

   Compiling cast v0.3.0
   Compiling hashlink v0.8.0
error[E0599]: no method named `powi` found for struct `f16` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-array-43.0.0/src/arithmetic.rs:351:25
    |
351 |                 Ok(self.powi(exp as i32))
    |                         ^^^^ method not found in `f16`
...
380 | native_type_float_op!(f16, f16::ZERO, f16::ONE);
    | ----------------------------------------------- in this macro invocation
    |
    = help: items from traits can only be used if the trait is in scope
    = note: this error originates in the macro `native_type_float_op` (in Nightly builds, run with -Z macro-backtrace for more info)
help: the following traits are implemented but not in scope; perhaps add a `use` for one of them:
    |
18  + use num::Float;
    |
18  + use num::traits::float::FloatCore;
    |
18  + use num::traits::real::Real;
    |
```

I'm on Rust 1.70.0, on MacOS.

Am I making a basic error?

**Comment by wangfenjin** [comment]
cargo clean and then retry?

**Comment by wangfenjin** [comment]
You are building this repo right?

**Comment by max-sixty** [comment]
> cargo clean and then retry?

This doesn't help, unfortunately.

I tried to compare to the CI build to understand the diff, but I notice that's disabled on Mac?

**Comment by wangfenjin** [comment]
I develop this crate in Mac, in order to save the CI time so I disable mac CI

**Comment by max-sixty** [comment]
Ok, any thoughts for resolving the build error then? Does a cargo clean work for you?

**Comment by wangfenjin** [comment]
It works fine for my side😂

**Comment by max-sixty** [comment]
`git clean -xfd` does work — probably it was an old `Cargo.lock`. `cargo clean` doesn't clear that.

Not sure if you want feedback on this... but #135 would probably fix this (but perhaps still waiting on upstream), and adding a CI check would make it very easy for you to tell people like me "it works in CI, it's your settings which are the problem"...

**Comment by wangfenjin** [comment]
Yes we are waiting arrow to fix the issue first
